src/dripper/polkadot/PolkadotActions.ts

```diff
@@ -112,16 +112,21 @@ export class PolkadotActions {
     // client.submit(tx) waits for the finalized value, which might be important for real money,
     // but for the faucet drips, early respond is better UX
     const submit$ = client.submitAndWatch(tx).pipe(shareReplay(1));
+    let hash = "";
+    submit$.subscribe({
+      next: (e) => {
+        hash = e.txHash;
+      },
+      error: (err) => {
+        if (hash) logger.error(`Transaction ${hash} failed to finalize`, err);
```

Comment:
Does it mean that `await firstValueFrom(...)` down below won't throw on errors?

Reply:
it will throw, of course.

However, please notice that the previous code was not taking into account the fact that it could throw prior to being broadcasted, and in those cases the `this.#pendingTransactions.delete(tx);` was not being done.

Reply:
basically, this subscription is the "equivalent" way of [doing this](https://github.com/paritytech/polkadot-testnet-faucet/pull/407/files#diff-801c8a922a2beb656a4bb54b0a2bb88c22032aa335dd983be04af6dc3a1b6282L118-L124), while also ensuring that the refcount doesn't go down to zero.

Reply:
Then in case of `logger.error(`Transaction ${tx} failed to broadcast`, err);`, the error will get double-logged?

Reply:
Oh, I see, you only want those logs to appear if an error happens after the promise of `sendTx` has already resolved. Gotcha! It's fixed now.